src/SimplePDPService.sol

```diff
@@ -105,6 +105,28 @@ contract SimplePDPService is PDPListener, PDPRecordKeeper, Initializable, UUPSUp
         return 2880;
     }
 
+    // Number of epochs at the end of a provoing period during which a 
+    // proof of possession can be submitted
+    function challengeWindow() public pure returns (uint256) {
+        return 60;
+    }
+
+    // The start of the next challenge window
+    // Useful for querying before nextProvingPeriod to determine challengeEpoch
+    function nextChallengeWindowStart(uint256 setId) public view returns (uint256) {
+        if (provingDeadlines[setId] == 0) {
+            revert("Proving period not yet open");
+        }
+        uint256 periodsSkipped;
+        // Proving period is open 0 skipped periods
+        if (block.number <= provingDeadlines[setId]) {
+            periodsSkipped = 0;
+        } else { // Proving period has closed possibly some skipped periods
+            periodsSkipped = (block.number - (provingDeadlines[setId] + 1)) / getMaxProvingPeriod();
+        }
+        return provingDeadlines[setId] + getMaxProvingPeriod()*(periodsSkipped+1) - challengeWindow();
```

Comment:
Please add tests for this method. Arithmetic like this is is tricky to mentally verify.

Reply:
@ZenGround0 Can you explain what we're doing here ? Where's this arithmetic coming from ?

Reply:
Ok I refactored this to be a lot less confusing.  The tricky thing is that if the current epoch falls before the latest marked deadline we want to roll to the *next* period's challenge window.  But if we've faulted, are late and the current epoch is well after the latest marked deadline we want to roll to the *current* period's window.  This is all now a lot more explicit.

Reply:
Yeah this all makes sense now 👍